www/front_src/src/Resources/Details/tabs/Graph/models.ts

```diff
@@ -4,39 +4,75 @@ import { find, propEq } from 'ramda';
 import { timeFormat, dateFormat } from '@centreon/ui';
 
 import {
-  labelLast24h,
-  labelLast7Days,
+  label24H,
+  label7D,
+  label31D,
+  label31Days,
   labelLast31Days,
+  label7Days,
+  labelLast7Days,
+  label24Hours,
+  labelLast24Hours,
 } from '../../../translatedLabels';
 
 export type TimePeriodId = 'last_24_h' | 'last_7_days' | 'last_31_days';
 
 export interface TimePeriod {
   id: TimePeriodId;
+  compactName: string;
   name: string;
+  largeName: string;
   getStart: () => Date;
   dateTimeFormat: string;
   timelineEventsLimit: number;
 }
 
+export interface CustomTimePeriod {
+  start: Date;
+  end: Date;
+  xAxisTickFormat?: string;
```

Comment:
```suggestion
  xAxisTickFormat?: string;
```
axisTickFormat

I propose this because for me the "x" and the word axis are two similar terms